boot-speed/clouds/ec2_cleanup.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env python3
+"""Clean up all running bootspeed-* EC2 instances.
+
+Copyright 2018-2019 Canonical Ltd.
+Joshua Powers <josh.powers@canonical.com>
+Paride Legovini <paride.legovini@canonical.com>
+"""
+import boto3
+import datetime as dt
+import re
+
+
+def clean_ec2():
+    """Clean up all running EC2 instances tagged 'bootspeed-*'."""
+    resource = boto3.resource('ec2')
+
+    inst_tag = [{'Name': 'tag:Name', 'Values': ['bootspeed-*']}]
+    for vpc in list(resource.vpcs.all()):
+        print('# cleaning up vpc %s\n' % vpc.id)
+        for instance in vpc.instances.filter(Filters=inst_tag):
+            for tag in instance.tags:
+                if tag['Key'] == 'Name' and re.match(
+                        "^bootspeed-[0-9]+-", tag['Value']):
+                    print('Found bootspeed instance %s' % instance.id)
+                    timestamp = int(tag['Value'].split("-")[1])
+                    tdelta = int(dt.datetime.utcnow().timestamp()) - timestamp
```

Comment:
I would suggest calculating "now" once so that the script has a consistent baseline that it's operating against; my past experience suggests that this makes behaviour easier to understand.